umich_api/api_utils.py

```diff
@@ -141,6 +172,12 @@ def get_access_token(self, token_url, client_scope):
         resp = requests.post(f"{self.base_url}/{token_url}", data=payload, headers=headers)
         try:
             if (resp.ok):
-                return resp.json().get('access_token')
+                r_json = resp.json()
+                # If expires_at is a parameter on the return, add the value to "now" and store it as a new value expires_time
+                if 'expires_at' in r_json:
+                    r_json['expires_time'] = datetime.now() + timedelta(seconds=r_json['expires_at'])
+                # Cache the token
+                self.tokens[token_url] = r_json
+                return r_json
         except (ValueError):
```

Comment:
if token generation leads to an expectation block, isn't this should be raising the exception to the calling method? that way actual caller which is application knows how to handles it. Currently, the way it seems to just return with our raising error to higher levels

Reply:
I don't remember what threw the ValueError. I guess the requests library could have a bad response too and then it couldn't parse the json. I'll change this to make it consistent with the other method.